src/devices_models/devices/common/range_constraint.jl

```diff
@@ -183,27 +191,31 @@ function reserve_device_semicontinuousrange(psi_container::PSIContainer,
 
     ub_name = _middle_rename(cons_name, "_", "ub")
     lb_name = _middle_rename(cons_name, "_", "lb")
-    names = collect(keys(range_data))
+    names = (d.name for d in range_data)
     #MOI has a semicontinous set, but after some tests is not clear most MILP solvers support it.
     #In the future this can be updated
     con_ub = add_cons_container!(psi_container, ub_name, names, time_steps)
     con_lb = add_cons_container!(psi_container, lb_name, names, time_steps)
 
-    for (name, data) in range_data
+    for data in range_data
```

Comment:
If the character count gets too long we can consolidate the for loops as 
```for t in time, data in range_data ``` and avoid an extra indentation.